storage/innobase/include/sync0policy.h

```diff
@@ -76,7 +76,7 @@ class MutexDebug {
 		{
 			m_mutex = mutex;
 
-			my_atomic_storelint(&m_thread_id, os_thread_get_curr_id());
+			my_atomic_storelint(&m_thread_id, (ulint)os_thread_get_curr_id());
```

Comment:
Could you change this either to `reinterpret_cast` or to `static_cast`?

Reply:
Sure, I tried to use the same style found later on line 160. Is it Ok to change that too or if it's unrelated to the actual patch leave it be?

Reply:
Line 160 is far from perfect due to defects in our atomics implementation. It is up to fix it or not, but please keep in mind every time we touch atomics, they break windows builds. We'll have to carefully pass this patch through buildbot before pushing.